### PR TITLE
Double check a non-empty but not None value was provided to jsoc notify

### PR DESCRIPTION
**Pull request by nabobalis**
This catches me out when running examples sometimes, as if you do not index the os.environ dict but use .get, you get a default value. I guess the answer is to fix the example but either way, we should catch this.

### CODE REVIEW

File: sunpy/net/jsoc/attrs.py [hunk 1]
```diff
@@ -110,9 +110,11 @@ class Notify(SimpleAttr):
 
     def __init__(self, value):
         super().__init__(value)
+        if not value:
+            raise ValueError("Notify attribute must contain an email address")
         if value.find('@') == -1:
             raise ValueError("Notify attribute must contain an '@' symbol "
-                             "to be a valid email address")
+                             "to be a valid email address: {value}")
```
Comment (nabobalis) [inline sunpy/net/jsoc/attrs.py:117]:
```suggestion
                             "to be a valid email address")
```

File: sunpy/net/jsoc/attrs.py [hunk 2]
```diff
@@ -110,9 +110,11 @@ class Notify(SimpleAttr):
 
     def __init__(self, value):
         super().__init__(value)
+        if not value:
```
Comment (dstansby) [inline sunpy/net/jsoc/attrs.py:113]:
```suggestion
        if value is None:
```
Is a bit clearer (if `value` is indeed `None` if no email is passed...

File: changelog/7342.trivial.rst [hunk 3]
```diff
@@ -0,0 +1 @@
+`~sunpy.net.jsoc.attrs.Notify` checks for a valid input value.
```
Comment (dstansby) [inline changelog/7342.trivial.rst:1]:
```suggestion
`~sunpy.net.jsoc.attrs.Notify` checks that a valid email address has been given as a value.
```
Does this have any other higher-level user facing behaviour that needs mentioning? e.g. if I forget to do a JSOC Fido search with an email address, does this now raise a more helpful error?

**Review by dstansby — CHANGES_REQUESTED** [review]
👍 , with a suggestion and question.

**Review by dstansby — APPROVED** [review]
Woops, that was meant to be an approval...

**Review by wtbarnes — APPROVED** [review]

**Pull request merged**